packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/util/CoroutineDispatcherFactory.kt

```diff
@@ -0,0 +1,63 @@
+package io.realm.kotlin.internal.util
+
+import io.realm.kotlin.internal.platform.multiThreadDispatcher
+import io.realm.kotlin.internal.platform.singleThreadDispatcher
+import kotlinx.coroutines.CloseableCoroutineDispatcher
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+/**
+ * Factory wrapper for passing around dispatchers without needing to create them. This makes it
+ * possible to configure dispatchers in Configuration objects, but allow Realm instances to control
+ * their lifecycle.
+ */
+public class CoroutineDispatcherFactory private constructor(
```

Comment:
We could also do that, I guess it feels slightly weird that the manager can close references it handed out without the dispatcher being involved, but at the same, it would help to get rid of the ManagedCoroutineDispatcher.

So I could probably agree that since both approaches have some annoyance, going with the one that has the least amount of code is nicer.

Reply:
Hmm, actually that is not going to work I think. the problem is that the `RealmConfiguration` can be used to open multiple Realm instances, all with their own dispatchers, so nothing inside the configuration is allowed to track handed-out configurations.

We would then end up with `RealmCoroutineFactoryFactory` ... which seems even more silly 🙈